fix: Minor edits in approach.adoc (#978)
starknet-io/starknet-docs: components/Starknet/modules/architecture_and_concepts/pages/Accounts/approach.adoc

A Starknet account contract must include the following two functions:

* `+__validate__+`
* `+__execute__+`

These functions serve distinct purposes to ensure that only the account owner can initiate transactions and that fees can be charged for the resources they use.

> Starknet's account type is inspired by Ethereum's EIP-4337, where instead of EOAs, you now use smart contract accounts with arbitrary verification logic.

Author: ```suggestion
Starknet's account type is inspired by Ethereum's EIP-4337, where instead of an account being an EOA, it is now a smart contract account with arbitrary verification logic.
```
